querydsl-jpa/src/main/java/com/querydsl/jpa/JPAQueryMixin.java

```diff
@@ -145,7 +145,8 @@ private boolean isEntityPath(Path<?> path) {
             Path<?> parent = shorten(metadata.getParent(), paths);
             Path<T> oldPath = new PathImpl<T>(path.getType(),
                     new PathMetadata(parent, metadata.getElement(), metadata.getPathType()));
-            Path<T> newPath = new PathImpl<T>(type, oldPath.toString().replace('.', '_'));
+            Path<T> newPath = new PathImpl<T>(type, ExpressionUtils.createRootVariable(oldPath));
```

Comment:
that's not available, so maybe do the change here https://github.com/querydsl/querydsl/pull/1285 ?